### ISSUE TITLE
Beethoven X - OPTIMISM QA

### ISSUE THREAD

**Issue by SerenityFund** [messari/subgraphs#1699]
## Review Info

| Description | Info | 
| :-------- | :--- | 
| Subgraph Reviewed |  https://subgraphs.messari.io/subgraph?endpoint=https://api.thegraph.com/subgraphs/name/messari/beethoven-x-optimism
| Schema Version  |  1.3.0  | 
| Subgraph Version |  1.1.2  | 
| Methodology Version  |	1.0.0 |
| Evidence Spreadsheet |  https://docs.google.com/spreadsheets/d/1Mjf9dPXHak54jcGSdnjT5G1CdW1wOrK7/edit?usp=share_link&ouid=104027705798814439445&rtpof=true&sd=true |

###  TVL

TVL data is not mathematically in line with the recent (Feb. 3) official website data as it shows a massive discrepancy of -50.51%. Additionally, in the five past five days, the TVL data shows discrepancies ranging from -46.20% to -57.78% with Defi Llama data,

February 3, 2023:

| Subgraph - TVL | Official Website - TVL | Absolute difference | Difference % 
| :---- | :--- | :---  | :---  | 
| 10,169,476| 20,550,000 | -10,380,524| -50.51%| 

| Subgraph - TVL | Defi Llama - TVL | Absolute difference | Difference % |
| :---- | :--- | :---  | :---  | 
| 10,169,476| 20,565,446 | -10,395,970| -50.55%| 

- Refer to the sheet 'Protocol'   

###  Revenue

The daily total revenue data is not aligned with the revenue data from the official website, showing a discrepancy of -28.17%.

February 3, 2023:

| Subgraph - Daily Total Revenue | Official Website - Daily Total Revenue | Absolute difference | Difference % 
| :---- | :--- | :---  | :---  | 
| 606| 844 | -238| -28.17%| 

- See the 'Protocol' sheet for more information.

###  Daily Volume

The recent daily volume data shows a high discrepancy of -28.29% with official website data.

February 3, 2023:

| Subgraph - Daily Volume | Official Website - Daily Volume | Absolute difference | Difference % 
| :---- | :--- | :---  | :---  | 
| 774,436| 1,080,000 | -305,564| -28.29%| 

### Daily Transaction Count

Daily transaction count data is not mathematically in line with the sum of deposit, withdraw, and swap daily counts in the past five days.

### Total Pool Count

Dexscreener, Defi Llama, and the official website only show 38, 92, and 9 pools respectively (compared to the subgraph's 142 pools).

### Pool Overview - TVL and Rewards

- Out of the top 20 pools in the subgraph data, only seven pools are active (shown in the website). 
- Five of those seven active pools' TVL data show large discrepancies with the official website data. These pools are:

| Subgraph - Rocket Fuel Pool TVL | Official Website - Rocket Fuel Pool TVL | Difference % 
| :---- | :--- | :---  | 
| $3,793,961| $6,160,973 | -38.42%| 

| Subgraph - Shanghai Shakedown Pool TVL | Official Website - Shanghai Shakedown Pool TVL | Difference % 
| :---- | :--- | :---  | 
| $1,934,181| $7,129,429 | -72.87%| 

| Subgraph - Happy Road Pool TVL | Official Website - Happy Road Pool TVL | Difference % 
| :---- | :--- | :---  | 
| $389,207| $326,906 | 19.06%| 

| Subgraph - Helter Skelter Pool TVL | Official Website - Helter Skelter Pool TVL | Difference % 
| :---- | :--- | :---  | 
| $137,525| $275,045 | -50.00%| 

| Subgraph - All You Need Is Love Pool TVL | Official Website - All You Need Is Love Pool TVL | Difference % 
| :---- | :--- | :---  | 
| $20,014| $99,592 | -79.90%| 

- Two pools from the official website are missing, namely 'Overnight Pulse' and 'Galactic Dragon' pools
- The tokens reward APR data on 'REWARD TOKENS' column is not mathematically in line with the official website data.
- In terms of the base yield, only one pool is showing a high discrepancy with the official website data which is the 'All You Need Is Love' pool
- Refer to the sheet "Pool_Overview" for details

Please refer to the attached worksheet for more details.

**Comment by bye43** [comment]
@harsh9200 QA here. Feels like those protocol level volume and TVL metrics are probably from that pricing issue we discussed.

**Comment by dmelotik** [comment]
@harsh9200 @bye43 Will this be fixed when we apply the uniswap fork pricing fix to balancer forks? 

cc @steegecs

**Comment by steegecs** [comment]
I think if we derived the prices from the pool it would. However, it looks like currently the prices are coming from harsh's price lib. Can you confirm @harsh9200?

**Comment by harsh9200** [comment]
Yes, right now balancer forks deployments are using my price lib. My price lib does not handle some tokens artificially created by balancer ([Example](https://arbiscan.io/address/0x4d00c6dd5d5299082a1062c9b480af2fc698f6eb)). To fix this I have deployed a price lib based on a new method which is deriving prices based on the swaps of a pool, but It has also some bugs that I am still fixing. I wonder if @steegecs price lib would work on these tokens?
